src/pages/api/comments.ts

```diff
@@ -0,0 +1,151 @@
+} from "../../types";
+
+const commentSelect = {
+  accidentId: true,
+  author: {
+    select: {
+      name: true,
+      email: true,
+      picture: true,
+    },
+  },
+  createdAt: true,
+  id: true,
+  isPublished: true,
+  text: true,
+} as const;
+
+const convertSelectedCommentToPublicInfo = ({
+  accidentId,
+  author,
+  createdAt,
+  id,
+  isPublished,
+  text,
+}: Prisma.CommentGetPayload<{
+  select: typeof commentSelect;
+}>): PublicCommentInfo => ({
+  accidentId,
+  authorAvatarUrl: generateAvatarUrl(author.picture ?? undefined, author.email),
+  authorName: author.name,
+  createdAt: createdAt.toISOString(),
+  id,
+  isPublished,
+  text,
+});
+
+const handler: NextApiHandler = async (req, res) => {
+  const session = getSession(req, res);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- external user type is `any`
+  const myId = session?.user.sub;
+
+  if (req.method === "POST") {
+    if (typeof myId !== "string") {
+      res.status(401).json({
+        error: "not_authenticated",
+        description:
+          "The user does not have an active session or is not authenticated",
+      });
+
+      return;
+    }
+
+    let newCommentPayload: NewCommentPayload | undefined;
+
+    try {
+      const { accidentId, text } = JSON.parse(req.body as string) as Record<
+        keyof NewCommentPayload,
+        unknown
+      >; // type casting is tolerable as we are inside try/catch
+
+      if (
+        typeof accidentId !== "string" ||
+        typeof text !== "string" ||
+        !accidentId ||
+        !text
+      ) {
+        throw new Error("Unexpected field types or fields are empty");
+      }
+
+      newCommentPayload = { accidentId, text };
+    } catch {
+      res.status(422).json({
+        error: "invalid_payload",
+        description: "New comment payload is corrupt and cannot be processed",
+      });
+
+      return;
+    }
+
+    const selectedComment = await prisma.comment.create({
+      data: {
+        ...newCommentPayload,
+        authorId: myId,
+      },
+      select: commentSelect,
+    });
+
+    const responseBody: CommentsApiHandlerSuccessfulPostResponseBody = {
+      status: "ok",
+      comment: convertSelectedCommentToPublicInfo(selectedComment),
+    };
+    res.status(200).json(responseBody);
+  } else if (req.method === "GET") {
+    const accidentId = req.query["accident-id"];
+
+    if (typeof accidentId !== "string" || !accidentId) {
+      res.status(422).json({
+        error: "missing_param",
+        description: "query param accident-id is missing",
+      });
+
+      return;
+    }
+    const selectedComments = await prisma.comment.findMany({
+      where: {
+        OR: [
+          {
+            isPublished: true,
+            accidentId,
+          },
+          ...(typeof myId === "string"
+            ? [
+                {
+                  accidentId,
+                  authorId: myId,
+                },
+              ]
+            : []),
```

Comment:
This will let us see own pending comments after page refresh. WDYT @sergous?

<img width="682" alt="Screenshot 2022-02-05 at 23 57 11" src="https://user-images.githubusercontent.com/608862/152663000-6ee167c4-cb7b-479d-a9cd-15af14154b49.png">

Reply:
Why not, they are just visible until refresh, so user can have some submit feedback. We can replace it with Text notice, explaining the moderation process and timing.

Reply:
They are shown _after_ refresh too now. That's because we now how `OR` in `where`.